test/context_test.js

```diff
@@ -108,4 +108,30 @@ suite('API (context)', function() {
       },
     });
   });
+
+  test('Context entry should be known', async () => {
+    //Create test api
+    var api = new subject({
+      title:        'Test Api',
+      description:  'Another test api',
+      context:      ['prop1', 'prop2'],
+    });
+
+    var value = slugid.v4();
+    let validate = await validator({
+      folder:         path.join(__dirname, 'schemas'),
+      baseUrl:        'http://localhost:4321/',
+    });
+    try {
+      api.router({
+        validator: validate,
+        context: {
+          prop3: 'value3',
+        },
+      });
+    } catch (err) {
+      return; //expected error
+    }
+    assert(false, 'Expected an error!');
+  });
```

Comment:
This is a good new test.  However, it's not catching the error you want it to catch!  If you modify the `catch` section like this:
```
    } catch (err) {
      if (/Context has unexpected property: 'prop3'/.test(err)) {
        return; //expected error
      }   
    }   
```

then the test will fail.  It would be helpful to also make a similar modification (but, looking for a different message) to the "Context properties can be required" test that you copied this one from.